geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneQueriesIntegrationTest.java

```diff
@@ -290,6 +295,24 @@ public void queryJsonObject() throws Exception {
   }
 
   @Test()
+  public void testWaitUntilFlushedForException() throws Exception {
+    Map<String, Analyzer> fields = new HashMap<String, Analyzer>();
+    fields.put("name", null);
+    fields.put("lastName", null);
+    fields.put("address", null);
+    luceneService.createIndexFactory().setFields(fields).create(INDEX_NAME, REGION_NAME);
+    Region region = cache.createRegionFactory(RegionShortcut.PARTITION).create(REGION_NAME);
+    final LuceneIndex index = luceneService.getIndex(INDEX_NAME, REGION_NAME);
+
+    // This is to send IllegalStateException from WaitUntilFlushedFunction
+    String nonCreatedIndex = "index2";
+
+    boolean b =
+        luceneService.waitUntilFlushed(nonCreatedIndex, REGION_NAME, 60000, TimeUnit.MILLISECONDS);
+    assertFalse(b);
```

Comment:
So the end result would be the wait for flush function would return false (it did not flush) and a message would be logged on the server side.  Based on the proposed fix of returning the exception, we would have been doing something similar anyways.  The client would have seen a false (because the function handled the exception and returned false)

So I guess having the test check for false would be just fine after all :-)